src/GameState/index.tsx

```diff
@@ -12,38 +12,29 @@ import {
   Spinner,
   Stack,
   Text,
+  useToast,
   View,
 } from 'native-base';
 import React, { useEffect, useReducer } from 'react';
 import { RootStackParamList } from '../../App';
 import Constants from 'expo-constants';
-import { Game, User } from '../interface';
+import { Game, GameInitialization, User } from '../interface';
 import { Board } from './Board';
 import { Marble } from './Marble';
 import { background } from 'styled-system';
 
 type Props = NativeStackScreenProps<RootStackParamList, 'GameState'>;
 const API_URL = Constants?.manifest?.extra?.API_URL;
 
-const initialState: {
-  players: User[];
-  game: Game | undefined;
-  isLoading: boolean;
-  error: any | undefined;
-} = {
+const initialState: GameInitialization = {
   players: [],
   game: undefined,
   isLoading: true,
   error: undefined,
 };
 
 const reducer = (
-  state: {
-    players: User[];
-    game: Game | undefined;
-    isLoading: boolean;
-    error: any | undefined;
-  },
+  state: GameInitialization,
```

Comment:
That wasn't really the point of my comment... I was talking more about something like 

```javascript
const GAME_INITIALIZATION = 'gameInitialization';
const reducer = (
  state: DefaultState,
  action: { type: string; value: any },
) => {
    switch (action.type) {
        case GAME_INITIALIZATION:
            return { ...state, game: action.value };
        default:
            return state;
    }
};
```

Then

```javascript
const initGame = game => dispatch({ type: GAME_INITIALIZATION, value: game });
```

But **if necessary** you can do it later